packages/content-type-reply/src/Reply.ts

```diff
@@ -0,0 +1,93 @@
+import { ContentTypeId } from "@xmtp/xmtp-js";
+import type {
+  CodecRegistry,
+  ContentCodec,
+  EncodedContent,
+} from "@xmtp/xmtp-js";
+import { composite as proto } from "@xmtp/proto";
+
+export const ContentTypeReply = new ContentTypeId({
+  authorityId: "xmtp.org",
+  typeId: "reply",
+  versionMajor: 1,
+  versionMinor: 0,
+});
+
+export type Reply = {
+  /**
+   * The message ID for the message that is being replied to
+   */
+  reference: string;
+  /**
+   * The content of the reply
+   */
+  content: any;
+  /**
+   * The content type of the reply
+   */
+  contentType: ContentTypeId;
+};
+
+export type ReplyParameters = Pick<Reply, "reference"> & {
+  contentType: string;
+};
+
+export class ReplyCodec implements ContentCodec<Reply> {
+  get contentType(): ContentTypeId {
+    return ContentTypeReply;
+  }
+
+  encode(
+    content: Reply,
+    codecs: CodecRegistry,
+  ): EncodedContent<ReplyParameters> {
+    const codec = codecs.codecFor(content.contentType);
+    if (!codec) {
+      throw new Error(
+        `missing codec for content type "${content.contentType.toString()}"`,
+      );
+    }
+
+    const bytes = proto.Composite.encode({
+      parts: [
+        {
+          part: codec.encode(content.content, codecs),
+          composite: undefined,
+        },
+      ],
+    }).finish();
```

Comment:
I'm not sure we need to use the Composite stuff here, I think it could just be something like:

```ts
import { content as proto } from "@xmtp/proto";
```

Then:

```ts
const encodedContent = codec.encode(content.content, codecs)
const bytes = proto.EncodedContent.encode(encodedContent).finish()
```

Then do something similar for decode.

Reply:
ok, i'll look into that. thanks!

Reply:
@nakajima the latest commit should address this